### ISSUE TITLE
Both sharing and exporting seem broken

### ISSUE THREAD

**Issue by dhouck** [ncase/simulating#7]
Chrome 66.0.3359.181/Windows 10

# Symptoms

Clicking "save your model" changes "When you save your model, you'll get a link here:" label to "saving...", and the "and an embed code here:" label and two textboxes to just "...". After waiting a while, they don't change to actual links or embed codes.

Clicking the "export model" button opens a new tab to about:blank.

# Probably explanation

I don't know if this is related to export not working correctly, but the linking is probably broken because Google is discontinuing goo.gl; see [this blog post](https://developers.googleblog.com/2018/03/transitioning-google-url-shortener.html). When I try saving, I get this message in console:

> FIREBASE WARNING: This Firebase project requires import into Google Cloud Platform to resume service. Visit the doc https://firebase.google.com/support/guides/firebase-web#import_your_project_to_the_firebase_console or reach out to Firebase support for help. Without action, this project will be permanently deleted on Tuesday, May 15th, 2018. (https://torrid-inferno-877.firebaseio.com)

You might note that it is now after May 15th, 2018. Fortunately, most of the application appears to have not been permanently deleted because most of it was not run from Google servers.

**Comment by andrewbellay** [comment]
@ncase - I can confirm this issue from @dhouck. I'm in Chrome, Version 68.0.3440.106 (Official Build) (64-bit) for MacOS High Sierra Version 10.13.6 (17G65).

Saving and embedding do not work and return a console warning: REBASE WARNING: Firebase error. Please ensure that you spelled the name of your Firebase correctly (https://torrid-inferno-877.firebaseio.com) 

Super cool simulation!!

**Comment by wizzwizz4** [comment]
I can confirm on Firefox ESR, Firefox Quantum and iOS 9 Safari; by this point I assume that it's not browser-specific.

**Comment by ncase** [comment]
Hi! Yeah I've been too busy on other projects to come back to this, but it's true, the Firebase database is down. I need to find either 1) another option that won't cost an arm and a leg, or 2) a way to save data _in_ the URL itself, like I did with LOOPY. However that'd take a week-ish of developer time. I may just remove the feature entirely for a while, until I get time to fix it. Thanks for your patience!

**Comment by wizzwizz4** [comment]
It's fine, because it failed with a loading bar instead of failing poorly. As an intermediate feature, you could do something like storing the JSON files in a (quota'd) directory on your server and having a query string load the JSON file from a built-up URL. There'd have to be client-side validation to stop XSS, but a quick-and-dirty solution wouldn't be too hard and could last for _months_ without anyone bugging you about it.

**Comment by ultimate-doge** [comment]
@ncase What about this, write every data into a single text file and send it to the user to download, this could work because text files aren't hard to work with. If this method seems too [INSERT WHATEVER ADJECTIVE YOU THINK IS RIGHT HERE], look far, there is also a lot of chinese services that will work just as fine, are they are very cheap, you will definitely not sell an arm or a leg or anything to do it. I found one that may work: [www.niaoyun.com](url)

**Comment by wizzwizz4** [comment]
@ncase We've got ourselves another spammer. :-/

**Comment by ultimate-doge** [comment]
Waddya mean @wizzwizz4

**Comment by wizzwizz4** [comment]
@ultimate-doge Never mind.

I thought you were a spammer because this was some of your first account activity and it looks like spam. But usually spammers don't respond months later... so I'm calling off the inquisition.

**Comment by ultimate-doge** [comment]
Yeah because I've made this account for a s[nope, can't swear]t load of time of time but don't really do anything, this is one of the things that I felt like I can help

**Comment by Openstreetmapler** [comment]
**Please fix this problem, I want to have fun doing complex things that cannot be done in one session and want to save my work!** By the way: clicking _export model_ doesn’t open an empty tab, instead a tab with the JSON text opens but **closes immediately after split seconds – there’s no time for copying the text!**

**Comment by Openstreetmapler** [comment]
> By the way: clicking _export model_ doesn’t open an empty tab, instead a tab with the JSON text opens but **closes immediately after split seconds – there’s no time for copying the text!**

Now, in 2022, with Firefox 107.0, the tab _does_ stay open, i. e. the Export Model feature isn’t broken anymore. But the saving still is.
